[c10d] Rename PG name and PG ID attribute

fduwjj opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #132915
* #132058

As discussed in https://github.com/pytorch/pytorch/pull/132058. we think pg_uid and local_uid might be a better name for pg_name and pg_id. So this PR is trying to rename it. More PRs are needed to change on the logging and other places.

cc @XilunWu @H-Huang @awgu @kwen2501 @wanchaol @fegin @wz337 @wconstab @d4l3k @c-p-i-o

fduwjj committed:
[c10d] Rename PG name and PG ID attribute

[ghstack-poisoned]

wconstab commented on torch/csrc/distributed/c10d/ProcessGroupNCCL.cpp:
```diff
@@ -775,13 +775,13 @@ ProcessGroupNCCL::ProcessGroupNCCL(
       terminateHeartbeatMonitorThread_(false),
       collectiveDebugInfoMode_(false),
-      uid_(process_group_id++),
+      local_uid_(process_group_id++),
```
Is local_id better? UID means unique id, it's arguably unique locally but adding the U might make people think it's more globally unique

wconstab commented on torch/csrc/distributed/c10d/ProcessGroupNCCL.cpp:
```diff
@@ -1593,16 +1593,17 @@ std::string ProcessGroupNCCL::createLogPrefix() const {
+        local_uid_,
         "PG GUID ",
-        pg_name_,
+        pg_uid_,
```
Is pg_global_uid better? I'm ok with uid too if we take u out of local 

Unlike local id which is a simple counter, this variable has special properties. I want to have a hint about that from the name.

fduwjj replied:
I prefer taking out u from local. adding global here seems a little bit duplicate.

fduwjj committed:
Update on "[c10d] Rename PG name and PG ID attribute"


As discussed in https://github.com/pytorch/pytorch/pull/132058. we think pg_uid and local_uid might be a better name for pg_name and pg_id. So this PR is trying to rename it. More PRs are needed to change on the logging and other places.

cc XilunWu H-Huang awgu kwen2501 wanchaol fegin wz337 wconstab d4l3k c-p-i-o

[ghstack-poisoned]

fduwjj commented:
@pytorchbot merge
